kimchi/src/circuits/gate.rs

```diff
@@ -208,7 +209,7 @@ impl<F: PrimeField> CircuitGate<F> {
         &self,
         row: usize,
         witness: &[Vec<F>; COLUMNS],
-        cs: &ConstraintSystem<F>,
+        index: &ProverIndex<G>,
```

Comment:
I really think we should pass the actual information we need, instead of passing the whole index or cs